modules/core/src/kmeans.cpp

```diff
@@ -458,7 +458,11 @@ double cv::kmeans( InputArray _data, int K,
         {
             best_compactness = compactness;
             if( _centers.needed() )
+            {
+                if(_centers.channels() == dims)
```

Comment:
It looks like the function allow to pass complex configuration of input points since in L239 `int dims = (isrow ? 1 : data0.cols)*data0.channels()`. IMHO it should be possible for the user to provide the same configuration for the `OutputArray _centers`

Reply:
This makes sense, will be implemented